src/types.ts

```diff
@@ -10,49 +10,54 @@ export const chainExplorerAPIKeySchema = z.object({
 
 export const chainExplorerAPISchema = z.object({
   key: chainExplorerAPIKeySchema,
-  url: z.string().url(),
+  url: z.url(),
 });
 
 export const chainExplorerSchema = z.object({
   api: chainExplorerAPISchema.optional(),
-  browserUrl: z.string().url(),
+  browserUrl: z.url(),
 });
 
 export const chainProviderSchema = z
   .object({
     alias: z.string(),
-    homepageUrl: z.string().url().optional(),
-    rpcUrl: z.string().url().optional(),
+    homepageUrl: z.url().optional(),
+    rpcUrl: z.url().optional(),
   })
   .refine(
     // Either rpcUrl or homepageUrl must be present
     (provider) => provider.rpcUrl ?? provider.homepageUrl,
-    { message: 'rpcUrl or homepageUrl is required' }
+    {
+      error: 'rpcUrl or homepageUrl is required',
+    }
   );
 
 export const chainProvidersSchema = z.array(chainProviderSchema).superRefine((providers, ctx) => {
   if (!providers.some((p) => p.alias === 'default')) {
-    ctx.addIssue({
-      code: z.ZodIssueCode.custom,
+    ctx.issues.push({
+      code: 'custom',
       path: ['providers', 'alias'],
       message: "a provider with alias 'default' is required",
+      input: providers.map((p) => p.alias),
     });
   }
 
   if (!hasUniqueEntries(providers, 'alias')) {
-    ctx.addIssue({
-      code: z.ZodIssueCode.custom,
+    ctx.issues.push({
+      code: 'custom',
       path: ['providers', 'alias'],
       message: "cannot contain duplicate 'alias' values",
+      input: providers.map((p) => p.alias),
     });
   }
 
   providers.forEach((p) => {
     if ((p.alias === 'default' || p.alias === 'public') && !p.rpcUrl) {
-      ctx.addIssue({
-        code: z.ZodIssueCode.custom,
+      ctx.issues.push({
+        code: 'custom',
         path: ['providers', 'rpcUrl'],
         message: "providers with alias 'default' or 'public' must also have an 'rpcUrl'",
+        input: p.rpcUrl,
```

Comment:
```suggestion
        path: [index, 'rpcUrl'],
        message: "providers with alias 'default' or 'public' must also have an 'rpcUrl'",
        input: p.rpcUrl,
```

If you'd like to follow my approach mentioned above.